### DISCUSSION TITLE
example codes with React.js

### DISCUSSION THREAD

**Discussion by dante01yoon** [mdn/mdn-community#538] [Content] [answered]
Hi, do I have any chances to  open pull request about providing code examples with React.js version?

May be it would helps a lot who use modern javascript framework.

**Comment by wbamberg** [comment, marked as answer]
Hi @dante01yoon ! Thanks for offering to help out MDN!

Documenting JS frameworks on MDN is tricky for a few reasons:

* MDN exists to document web standards and these are not web standards. If we document framework A, people will always ask why we do not document frameworks B, C, ...Z as well.
* Frameworks tend to change (and go in and out of fashion) more quickly than web standards, so maintaining the docs is more challenging. MDN is huge and maintenance is a big issue.
* Frameworks typically have their own excellent documentation. What's the value in MDN having its own version, that will usually be less up to date?

That said, they're obviously a big part of modern web development, so we do have some documentation of JavaScript frameworks here: https://developer.mozilla.org/en-US/docs/Learn/Tools_and_testing/Client-side_JavaScript_frameworks and would welcome help maintaining this. But I don't think we should start documenting (say) React in the main JS documentation.

**Reply by dante01yoon** [reply]
Wow, I didn't expect that you guys answer my discussion that kindly.
All the reasons you stated are good enough to convince me. 
thankyou!

**Reply by dante01yoon** [reply]
If the people who use each framework ported the MDN documentation and provided it, wouldn't the maintenance cost be greatly reduced?
What do you think about this? 

I had to change vanilla javascript codes in to React when I found web-RTC standard examples in MDN doc. Although these web standards should be treated as official approach, I wonder framework users can be helped enough when they build their own programs.

**Comment by pepelsbey** [comment]
Closing as answered/settled.
